### PR TITLE
[BugFix] Fix potential bug and refine log printing

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Comment by sduzh** [comment]
run starrocks_be_build

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 7 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|7|00.00%|[61, 71, 72, 74, 78, 79, 80]|

File: be/src/storage/lake/tablet_manager.cpp [hunk 1]
```diff
@@ -575,8 +577,12 @@ void* metadata_gc_trigger(void* arg) {
                 auto r = metadata_gc(root, tablet_mgr);
                 LOG_IF(WARNING, !r.ok()) << "Fail to do metadata gc in " << root << ": " << r;
             });
-            LOG_IF(WARNING, !st.ok()) << "Fail to submit task to threadpool: " << st;
+            if (!st.ok()) {
+                LOG(WARNING) << "Fail to submit task to threadpool: " << st;
+                break;
```
Comment (tracymacding) [inline be/src/storage/lake/tablet_manager.cpp:582]: what would happen if continue to next root?

Comment (sduzh) [reply be/src/storage/lake/tablet_manager.cpp:582]: Nothing go wrong but has high possibility to fail again.

**Review by tracymacding — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by wyb — APPROVED** [review]

**Comment by wyb** [comment]
run starrocks_admit_test

**Pull request merged**